code/modules/projectiles/gun.dm

```diff
@@ -80,6 +81,23 @@
 	var/noricochet = FALSE // wether or not bullets fired from this gun can ricochet off of walls
 	var/inversed_carry = FALSE
 
+/obj/item/weapon/gun/attackby(obj/item/I, mob/living/user, params)
+	if(!istool(I))
+		return FALSE
+	if(!gun_parts)
+		to_chat(user, SPAN_NOTICE("You can't dismantle [src] it has no gun  parts!"))
```

Comment:
If this message appears it probably means something went horribly wrong and admins should be notified
```suggestion
		to_chat(user, SPAN_NOTICE("You can't dismantle [src] as it has no gun parts! How strange..."))
```